apps/platform/src/app/auth/page.tsx

```diff
@@ -51,6 +52,49 @@ export default function AuthPage(): React.JSX.Element {
     }
   }
 
+  const handleGithubLogin = (): void => {
+    setIsLoading(true)
+    window.location.href = `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/auth/github`
+  }
+
+  const handleGithubRedirect = async () => {
+    const urlParams = new URLSearchParams(window.location.search)
+    const code = urlParams.get('code')
+
+    if (code) {
+      try {
+        const response = await fetch(
+          `${process.env.NEXT_PUBLIC_BACKEND_URL}/api/auth/github/callback?code=${code}`,
+          {
+            method: 'GET',
+            credentials: 'include'
+          }
+        )
+
+        if (!response.ok) {
+          throw new Error('Network response was not ok')
+        }
+
+        const data = (await response.json()) as User
+        setEmail(data.email)
+
+        if (response.status === 200) {
+          router.push('/auth/account-details')
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console -- we need to log the error
+        console.error(`Failed to fetch user data: ${error}`)
```

Comment:
If you could add toast messages to handle the error states in a well-mannered way, then that would be really nice.

Reply:
Yeah, I planned to do this after implementing the functionalities first, but then I got stuck and someone asked me to send the PR for review. Is it working fine on your end?

Reply:
> Yeah, I planned to do this after implementing the functionalities first, but then I got stuck and someone asked me to send the PR for review. Is it working fine on your end?

What's the problem you are facing?

Reply:
this is getting redirected to the incorrect url

Reply:
> After clicking the button, the user is redirected to a GitHub auth screen. Then, an API request is made to fetch user data along with a token. I set the email state and redirect the user to the /account-details page, where everything works fine up to this point
> 
> ![image](https://private-user-images.githubusercontent.com/102988552/382727597-207fee31-6da6-474c-aaa1-9ad36afd4fe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNzAzNzgsIm5iZiI6MTczMTE3MDA3OCwicGF0aCI6Ii8xMDI5ODg1NTIvMzgyNzI3NTk3LTIwN2ZlZTMxLTZkYTYtNDc0Yy1hYWExLTlhZDM2YWZkNGZlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwOVQxNjM0MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MWRmZmU2NjE5NWNlYjc4Mzk2YjgxOTlmYjJhYTU1N2NjYjI1NzM1YWUwMzA2NWYzYzUwNmJiMzQyYjgxMDA2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.jWRM9kWkWzhkao1nD876jqghHr6TaE4_MKXv7aszFzA)
> 
> the response includes this message
> 
> ![image](https://private-user-images.githubusercontent.com/102988552/382727823-4a7e7166-3366-461d-a6e8-9b0243bd3d8e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzExNzAzNzgsIm5iZiI6MTczMTE3MDA3OCwicGF0aCI6Ii8xMDI5ODg1NTIvMzgyNzI3ODIzLTRhN2U3MTY2LTMzNjYtNDYxZC1hNmU4LTliMDI0M2JkM2Q4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwOVQxNjM0MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NTE3YzljY2EyOTY4ZjBkM2E4Nzg4ZGUwODcwZmZkNjU0YWMzN2U2YTAzYjgyZTI4OTk0NTQxMDA5YTA2MWZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.N--hT-N8B69aHobbygL0Qso2YGKJiHoPYI0FqqxRe0U)
> 
> If you could explain what it means, then maybe I could do something. Otherwise, it's a dead end for me :)

this is the problem